test/libsolidity/astPropertyTests/forLoop/isSimpleCounterLoop.sol

```diff
@@ -0,0 +1,29 @@
+contract C {
+  function f() public pure {
+     /// PositiveCase1: isSimpleCounterLoop
+     for(uint i = 0; i < 42; ++i) {
+     }
+     /// PositiveCase2: isSimpleCounterLoop
+     for(int i = 0; i < 42; i++) {
+     }
+     /// NegativeCase1: isSimpleCounterLoop
+     for(uint i = 0; i < 42; i = i + 1) {
+     }
+     /// NegativeCase2: isSimpleCounterLoop
+     for(uint i = 42; i > 0; --i) {
+     }
+     /// NegativeCase3: isSimpleCounterLoop
+     for(uint i = 42; i > 0; i--) {
+     }
+     /// NegativeCase4: isSimpleCounterLoop
+     for(uint i = 1; i < 42; i = i * 2) {
+     }
+  }
+}
+// ----
+// PositiveCase1: true
+// PositiveCase2: true
+// NegativeCase1: false
+// NegativeCase2: false
+// NegativeCase3: false
+// NegativeCase4: false
```

Comment:
I still don't see that - if you put the path in the AST in the expectation, you need to jump from the expectation to the code to trace down the subpath, it's just a different kind of jumping. Especially, since the hook point for the AST paths are on statement level and in the future we may want to check properties on the expression level.